ding/utils/memory_helper.py

```diff
@@ -0,0 +1,722 @@
+from typing import Any, Dict, List, Tuple
+from collections import OrderedDict
+from functools import partial, reduce
+
+import os
+import time
+import torch
+try:
+    import pyecharts
+except ImportError:
+    import sys
+    import logging
+    logging.error("Please install pyecharts first, you can install it by running 'pip install pyecharts'")
+    sys.exit(1)
+
+mb = 1024 * 1024
+
+
+class SimpleMemState:
+    """
+    A class to represent the memory state of a model layer.
```

Comment:
Overview:
            A class to represent the memory state of a model layer.
        Arguments:
            - layer_name (:obj:`str`): The name of the layer.
            - layer_mem (:obj:`int`): The memory usage of the layer in bytes.